app/helpers/quadicon_helper.rb

```diff
@@ -461,7 +461,7 @@ def render_service_quadicon(item, options)
     link_opts = {}
 
     if quadicon_show_links?
-      url = options[:url] || quadicon_url_to_xshow_from_cid(item, options)
+      quadicon_url_to_xshow_from_cid(item, options)
```

Comment:
missing `url =`?

Reply:
The `url` isn't necessary.

The routing happens with -

```ruby
{:action => 'x_show', :id => to_cid(item.id) || options[:x_show_id]}
```
which is what the `quadicon_url_to_xshow_from_cid` method is all about.

Reply:
@himdel : I think you are right. I deleted extra `url =`. 

It does not matter because it the URL is empty then the URL provided in the ReportDataController is used. But we should fix this anyway --> doing it.

@AparnaKarve : The returned value, the 
```{:action => 'x_show', :id => to_cid(item.id) || options[:x_show_id]}```
is ignored. What do you mean by your comment? Can you rephrase that?

Reply:
@martinpovolny @himdel Never mind ...took a second look and we *do* need it to be like this -

```
url = quadicon_url_to_xshow_from_cid(item, options)
```

Reply:
@martinpovolny Adding `url` to this line has created a routing issue.

Log details are here - https://github.com/ManageIQ/manageiq-ui-classic/pull/2794#issuecomment-346498425

Fixing this in routes.rb should be simple.
(Need a good spec to go with it though.)